mwdb/model/file.py

```diff
@@ -279,6 +280,38 @@ def iterate(self, chunk_size=1024 * 256):
         finally:
             File.close(fh)
 
+    def xor(self, chunk):
```

Comment:
1. This is technically `not`, and not `xor` - key is not parametrised.

2. Pulling in numpy as a dependency for something as simple as xor is probably unwarranted. Please reimplement this function yourself. Out of the top of my head this should work:

```python
def negate_bits(self, chunk: bytes) -> bytes:
    return bytes([b ^ 0xFF for b in chunk])
```

Reply:
Hi, thanks for your review!
In refer to your comment, at first I was using solution similar to yours, but then I considered it as bottleneck. Thats why I applied numpy solution, which is about 50 times faster for this case, what could make difference for bigger files. I know that another dependency is something unwanted, but I think it should be at least considered. I attach code for timing tests

```python
import timeit
import numpy as np

def negate_bits(chunk):
    return bytes([b ^ 0xFF for b in chunk])

def negate_bits_numpy(chunk):
    key = np.frombuffer(b'\xff'*len(chunk), dtype='uint8')
    chunk = np.frombuffer(chunk, dtype='uint8')
    return (key ^ chunk).tobytes()

chunk = b'A'*1024*1024*1  # 1MB
print(timeit.timeit(lambda: negate_bits(chunk), number=10))
print(timeit.timeit(lambda: negate_bits_numpy(chunk), number=10))
```

Reply:
I thought exactly the same about adding heavy dependency just to make a xor operation.

But now I see it's actually much slower than expected e.g. on my laptop:
```
import timeit
import numpy as np

def negate_bits(chunk):
    return bytes([b ^ 0xFF for b in chunk])

def negate_bits_numpy(chunk):
    key = np.frombuffer(b'\xff'*len(chunk), dtype='uint8')
    chunk = np.frombuffer(chunk, dtype='uint8')
    return (key ^ chunk).tobytes()

chunk = b'A'*1024*1024*1  # 1MB
print(timeit.timeit(lambda: negate_bits(chunk), number=10))
print(timeit.timeit(lambda: negate_bits_numpy(chunk), number=10))

0.6028959036993911
0.009339740300492849
```
for 10MB file it's ~600ms vs ~3ms, so download will be possibly two times slower. What's even more interesting: `bytes(b ^ 0xFF for b in chunk)` is even slower than `bytes([b ^ 0xFF for b in chunk])`.

So maybe numpy should be actually considered in this case?

Reply:
Yeah, fair. That immediately caught my eye as an unnecessary dependency (looks like you saw that too), but I didn't realise it's *that* slow for large files.

If we can't get it done fast with pure python 100% fine to use the right tool for the job.